x/evm/types/state_transition.go

```diff
@@ -145,7 +145,8 @@ func (st StateTransition) TransitionDb(ctx sdk.Context, config ChainConfig) (*Ex
 	// This gas meter is set up to consume gas from gaskv during evm execution and be ignored
 	currentGasMeter := ctx.GasMeter()
 	evmGasMeter := sdk.NewInfiniteGasMeter()
-	csdb.WithContext(ctx.WithGasMeter(evmGasMeter))
+	ctx = ctx.WithGasMeter(evmGasMeter)
+	csdb.WithContext(ctx)
 
```

Comment:
It should add  
```
defer func(){ctx = ctx.WithGasMeter(currentGasMeter)}()
```
to recover currentGasMeter to ctx after executing evm actions.